app/src/main/java/toanns/traviscontrol/MainActivity.java

```diff
@@ -1,13 +1,100 @@
 package toanns.traviscontrol;
 
-import android.support.v7.app.AppCompatActivity;
+import android.os.AsyncTask;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.TextView;
 
 public class MainActivity extends AppCompatActivity {
 
+    private ProgressBar mProgressBar;
+    private Button mDownload;
+    private TextView mTvPercent;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        mDownload = (Button) findViewById(R.id.btnDownload);
+        mProgressBar = (ProgressBar) findViewById(R.id.progressbar);
+        mTvPercent = (TextView) findViewById(R.id.tvPercent);
+
+        mDownload.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+//                Download task = new Download();
+//                task.execute();
+                startProgress(v);
+            }
+        });
+
+    }
+
+    public void startProgress(View view) {
+        mProgressBar.setProgress(0);
+        new Thread(new Task()).start();
+    }
+
+    class Task implements Runnable {
+
+        @Override
+        public void run() {
+            for (int i = 0; i <= 100; i++) {
+
+                final int value= i;
+                try {
+                    Thread.sleep(200);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        mTvPercent.setText(value+"");
```

Comment:
WARNING: [SetTextI18n] Do not concatenate text displayed with `setText`. Use resource string with placeholders.
 When calling `TextView#setText`
* Never call `Number#toString()` to format numbers; it will not handle fraction separators and locale-specific digits properly. Consider using `String#format` with proper format specifications (`%d` or `%f`) instead.
* Do not pass a string literal (e.g. "Hello") to display text. Hardcoded text can not be properly translated to other languages. Consider using Android resource strings instead.
* Do not build messages by concatenating text chunks. Such messages can not be properly translated.